src/main/java/frc/robot/subsystems/GameObjectManager.java

```diff
@@ -165,13 +163,36 @@ public int bestPlace() {
 
     @Override
     public void periodic() {
-        Collection<Detection> detections = m_coprocessor.getAllDetections();
-        for (Detection d : detections) {
-            Pose2d pos = new Pose2d(d.getPosition().x, d.getPosition().y, new Rotation2d(0.));
-            Pose2d globalpos = pos.transformBy(new Transform2d(m_coprocessor.getTagGlobalPose(), new Pose2d()));
-            addGameObject(d.getName(), globalpos.getX(), globalpos.getY(), d.getPosition().z, 0);
+        if (m_coprocessor.isTagGlobalPoseActive()) {
+            Collection<Detection> detections = m_coprocessor.getAllDetections();
+            for (Detection d : detections) {
+                Pose2d pos = new Pose2d(d.getPosition().x, d.getPosition().y, new Rotation2d(0.));
+                Pose2d globalpos = pos.transformBy(new Transform2d(m_coprocessor.getTagGlobalPose(), new Pose2d()));
+                addGameObject(d.getName(), globalpos.getX(), globalpos.getY(), d.getPosition().z, 0);
+            }
+            removeInactiveGameObjects();
+            fillGridZones();
```

Comment:
Instead of checking every place in the grid, you could just check the column that's closest to the robot. That'll save a lot of compute. Especially since the drivers aren't using the full grid's info.

Reply:
I know we said we wanted to detect links, but for an initial release, we should focus on getting the minimum requirements fulfilled. I heard performance was an issue, so let's focus on making this code as lean as possible.